micropipenv.py

```diff
@@ -734,6 +744,17 @@ def _poetry2pipfile_lock(
         else:
             raise PoetryError("Unknown category for package {}: {}".format(entry["name"], entry["category"]))
 
+    for dependency_name, markers in additional_markers.items():
+        if dependency_name in default:
+            category = default
+        elif dependency_name in develop:
+            category = develop
+
+        if category[dependency_name].get("markers", None):
+            category[dependency_name]["markers"] += " or " + markers
```

Comment:
It might be a good idea to use brackets as markers can group expressions using `and`/`or`.

Reply:
As I'm thinking about this more, I'm not sure that this case might even happen. Let's say my app depends on twisted. Twisted depends on twisted-iocpsupport but only on windows. In this case, the markers are transferred from the definition of the dependency to the package definition and there is nothing to merge.
If my app for some reason depends on the twisted-iocpsupport directly, I'm responsible for setting the appropriate markers and because the dependency is direct, markers from the twisted dependency specification are not transferred to the package definition because twisted depends on twisted-iocpsupport only on windows but I specified it as a direct dependency and I might want to install it also on Linux for whatever reason and markers defined in twisted should not block me.

So, I'd say that we can remove that condition because if the dependency is not direct, there are no markers to merge the additional markers with.

What do you think?

Reply:
I think this case is relevant if a dependency is introduced by two different packages - not direct dependencies, but both are transitive:

A depends on B that depends on X
C depends on D that depends on X

If B and D define environment markers on X, they need to be merged using `or` as X needs to be installed if any from B and D need it.

Reply:
Thanks for the explanation. It makes perfect sense and I've also found a combination of projects where I can simulate this. Unfortunately, it also discovered a bug in my implementation because it does not normalize package names and fails if one package depends on `importlib-metadata` while another depends on `importlib_metadata` - they should be both normalized to `importlib-metadata` as described in PEP 503.

Reply:
Logic reimplemented and it now works with the described scenario with transient dependencies where they all have some markers - see the latest test.